src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -92,6 +92,30 @@ End Class
 ");
         }
 
+        [Fact]
+        public async Task CA2200_NoDiagnosticsForThrowAnotherExceptionInWhenClauseWithoutVariableDeclarator()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+public abstract class C
+{
+    public void M()
+    {
+        try
+        {
+        }
+        catch (Exception) when (Map(out Exception ex))
```

Comment:
@Youssef1313 please add both tests - one in the repro and one that you added here based on PR feedback. Sorry if my comment was unclear.

Reply:
@mavasani My bad. I thought I copied the existing test and modified the copy.